### ISSUE TITLE
Setting the same name parameters get the wrong error message saying `argument` instead of `parameter`

### ISSUE THREAD

**Issue by hyperkai** [python/mypy#20936]
*Notes:
- mypy test.py
- mypy 1.19.1
- Python 3.14.0

Setting the same name parameters get the wrong error message saying `argument` instead of `parameter` as shown below:

```python
       # ↓       ↓
def func(x: int, x: str): ...
# Error
```

> error: Duplicate argument "x" in function definition

So, the error message should say `parameter` as shown below:

> error: Duplicate parameter "x" in function definition
